src/test/test.cpp

```diff
@@ -251,10 +261,18 @@ runTest(CommandLineArgs const& args)
     {
         gVersionsToTest.emplace_back(Config::CURRENT_LEDGER_PROTOCOL_VERSION);
     }
+    if (seed != 0)
```

Comment:
I *think* a better way (or in addition to) is to use https://github.com/catchorg/Catch2/blob/master/docs/event-listeners.md#top
and set the seed every time we run a test or do something clever when sections start.

Without this, we still have non deterministic behavior of some tests (that depend on real clock) that bleed into other tests

Reply:
IF it's not already the case, we can do the same thing to `std::rand()`

Reply:
Oh, interesting! I will take a look, thanks!

Reply:
Fixed.